recipe/data.patch

```diff
@@ -1,18 +1,35 @@
---- lib/mpl_toolkits/basemap/__init__.py	2016-03-23 10:25:48.150934435 -0300
-+++ lib/mpl_toolkits/basemap/__init__.py	2016-03-23 14:22:02.486302483 -0300
-@@ -45,7 +45,10 @@
-     if not os.path.isdir(basemap_datadir):
-         raise RuntimeError('Path in environment BASEMAPDATA not a directory')
+--- basemap-1.2.1rel.orig/lib/mpl_toolkits/basemap/__init__.py	2019-08-07 22:03:01.000000000 -0300
++++ basemap-1.2.1rel/lib/mpl_toolkits/basemap/__init__.py	2019-08-08 11:15:02.145868394 -0300
+@@ -53,15 +53,10 @@
+ import _geoslib
+ import functools
+ 
+-# basemap data files now installed in lib/matplotlib/toolkits/basemap/data
+-# check to see if environment variable BASEMAPDATA set to a directory,
+-# and if so look for the data there.
+-if 'BASEMAPDATA' in os.environ:
+-    basemap_datadir = os.environ['BASEMAPDATA']
+-    if not os.path.isdir(basemap_datadir):
+-        raise RuntimeError('Path in environment BASEMAPDATA not a directory')
++if os.name == 'nt':
++    basemap_datadir = os.path.join(sys.prefix, 'Library', 'share', 'basemap')
  else:
 -    basemap_datadir = os.sep.join([os.path.dirname(__file__), 'data'])
-+  if os.name == 'nt':
-+    basemap_datadir = os.path.join(sys.prefix, 'Library', 'share', 'basemap')
-+  else:
 +    basemap_datadir = os.path.join(sys.prefix, 'share', 'basemap')
-
- __version__ = '1.0.8'
-
-@@ -1205,11 +1208,12 @@
+ 
+ __version__ = '1.2.1'
+ 
+@@ -154,7 +149,8 @@
+              }
+ 
+ # create dictionary that maps epsg codes to Basemap kwargs.
+-epsgf = open(os.path.join(basemap_datadir, 'epsg'))
++pyproj_datadir = os.environ['PROJ_LIB']
++epsgf = open(os.path.join(pyproj_datadir,'epsg'))
```

Comment:
I don't think this is needed anymore. basemap now packages the epsg file so that it can load it up on its own instead of depending on pyproj to provide it like it used to.

Reply:
Looks like it relies on an environment variable and falls back to a `data` directory, right? I'm looking at https://github.com/matplotlib/basemap/blob/master/lib/mpl_toolkits/basemap/__init__.py#L56-L64

So... we either need to set up that environment variable with an activation script, but that seems confusing b/c it is both `proj4` data and `basemap` data in a single directory, or we need to tell `basemap` to use the `conda-forge` `proj4` data like in this patch. I think I prefer the latter b/c we like to un-bundle things and makes it easier to debug problems, like why I have this version of `proj4` but `basemap` is using another one?

Reply:
Here is the PR that changed this for compatibility with pyproj v2: https://github.com/matplotlib/basemap/pull/454

I don't have a lot of bandwidth for figuring out what is the "right" solution. This probably does work, but I have a nagging feeling in the back of my head that we tried this and it didn't always work.

Reply:
That PR uses a bundled `proj4` data file that does not exist in `proj4 >=6`. It is not really related to `pyproj 2`. It does circumvent issues when `pyproj 2` was built with `proj4 >=6` though. However, like I mentioned above, users will be quite confused by having one `proj4` version installed but `basemap` is actually using data from another version.

>  but I have a nagging feeling in the back of my head that we tried this and it didn't always work.

Not sure what you mean. If you are referring to the patch issues most of them were problems with people misusing conda packages.